test/libsolidity/util/TestFileParser.cpp

```diff
@@ -571,8 +571,73 @@ string TestFileParser::Scanner::scanString()
 
 	while (current() != '\"')
 	{
-		str += current();
-		advance();
+		if (current() == '\\')
+		{
+			advance();
+			switch (current())
+			{
+				case '\\':
+					str += current();
+					advance();
+					break;
+				case 'n':
+					str += '\n';
+					advance();
+					break;
+				case 'r':
+					str += '\r';
+					advance();
+					break;
+				case 't':
+					str += '\t';
+					advance();
+					break;
+				case '0':
+					str += '\0';
+					advance();
+					break;
+				case 'x':
+					str += scanHexPart();
+					break;
+				default:
+					str += "\\";
+					str += current();
+					advance();
+					break;
+			}
+		}
+		else
+		{
+			str += current();
+			advance();
+		}
 	}
 	return str;
 }
+
+char TestFileParser::Scanner::scanHexPart()
+{
+	advance(); // skip 'x'
+
+	char value{};
+	if (isdigit(current()))
+		value = current() - '0';
+	else if (tolower(current()) >= 'a' && tolower(current()) <= 'f')
+		value = tolower(current()) - 'a' + 10;
+	else
+		throw Error(Error::Type::ParserError, "\\x used with no following hex digits.");
+
+	advance();
+	if (current() == '"')
```

Comment:
Shouldn't this be an error?

Reply:
No, we actually checked that with what the C/C++ compiler does, which taught us how to treat hex escape sequences at the end of strings. I map the syntax 1:1 and added tests to ensure this. :)